[Bug]: stackoverflowerror loading session

### Brokk Version

master

### Operating System

Linux

### What happened?

java.util.concurrent.ExecutionException: java.lang.StackOverflowError
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[?:?]
	at io.github.jbellis.brokk.SessionManager.loadHistory(SessionManager.java:447) ~[main/:?]
	at io.github.jbellis.brokk.gui.HistoryOutputPanel.lambda$triggerAiCountLoad$75(HistoryOutputPanel.java:2559) ~[main/:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188) [?:?]
Caused by: java.lang.StackOverflowError
	at java.lang.String.<init>(String.java:1425) ~[?:?]
	at io.github.jbellis.brokk.util.HistoryIo$ContentReader.readContent(HistoryIo.java:488) ~[main/:?]
	at io.github.jbellis.brokk.util.HistoryIo$ContentReader.readContent(HistoryIo.java:492) ~[main/:?]
	at io.github.jbellis.brokk.util.HistoryIo$ContentReader.readContent(HistoryIo.java:492) ~[main/:?]
[repeats]

### Relevant log output

```shell

```

--- a/app/src/main/java/io/github/jbellis/brokk/SessionManager.java
+++ b/app/src/main/java/io/github/jbellis/brokk/SessionManager.java
@@ -220,7 +220,7 @@ public class SessionManager implements AutoCloseable {
                 // Exercise migrations and quarantine if history read fails.
                 try {
                     loadHistoryOrQuarantine(sessionId, contextManager);
-                } catch (IOException e) {
+                } catch (Exception e) {
                     quarantinedIds.add(sessionId);
                     moved++;
                     continue;
@@ -339,12 +339,12 @@ public class SessionManager implements AutoCloseable {
         }
     }
 
-    private ContextHistory loadHistoryOrQuarantine(UUID sessionId, IContextManager contextManager) throws IOException {
+    private ContextHistory loadHistoryOrQuarantine(UUID sessionId, IContextManager contextManager) throws Exception {
         try {
             return loadHistoryInternal(sessionId, contextManager);
-        } catch (IOException e) {
+        } catch (Exception | StackOverflowError e) {
             moveSessionToUnreadable(sessionId);
-            throw e;
+            throw new Exception("Cannot read session history.", e);
         }
     }
 

